Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -739,10 +749,22 @@ std::unique_ptr<VKStagingTexture> VKStagingTexture::Create(StagingTextureType ty
     return nullptr;
   }
 
+  // Linear image
+  VkImage linear_image = VK_NULL_HANDLE;
+  VkDeviceMemory linear_image_device_memory = VK_NULL_HANDLE;
+  if (DriverDetails::HasBug(DriverDetails::BUG_SLOW_OPTIMAL_IMAGE_TO_BUFFER_COPY) &&
+      type == StagingTextureType::Readback && config.samples == 1)
+  {
+    auto [image, image_memory] = CreateLinearImage(type, config);
+    linear_image = image;
+    linear_image_device_memory = image_memory;
+  }
+
   std::unique_ptr<StagingBuffer> staging_buffer =
       std::make_unique<StagingBuffer>(buffer_type, buffer, memory, buffer_size, coherent);
-  std::unique_ptr<VKStagingTexture> staging_tex = std::unique_ptr<VKStagingTexture>(
-      new VKStagingTexture(type, config, std::move(staging_buffer)));
+  std::unique_ptr<VKStagingTexture> staging_tex =
+      std::unique_ptr<VKStagingTexture>(new VKStagingTexture(
+          type, config, std::move(staging_buffer), linear_image, linear_image_device_memory));
```

Comment:
Any idea why this isn't using `std::make_unique<VKStagingTexture>`? It seems like it should work just fine. (The texture code on the other backends also uses `new` in similar places; I'm not sure why.)

Reply:
I've noticed a number of places not using `make_unique`.  My guess is it wasn't available at one time?  I almost suggested this in this PR but figured it'd be better to fix them in one fell swoop.

Reply:
Both the line for `staging_buffer` and `staging_tex` were added in the same commit (f43d85921dc85ae586d69b21d28a7578835a55bd), so it's not a case of availability, but maybe there was some other language limitation at the time? (In any case, yeah, changing them all at once is probably better.)

Reply:
The constructor used there is private, so make_unique doesn't work.

Reply:
Yeah, this is usually handled by making the constructor public and having an internal tag that only the class knows how to create.  We have such an example in [BalloonTip](https://github.com/dolphin-emu/dolphin/blob/ffdc8538a162b1ca56e3943c5a78c02358053b2a/Source/Core/DolphinQt/Config/Graphics/BalloonTip.cpp#L52).

Reply:
Oh, I didn't know about that trick. I've just used `new` when I've ran into this situation in DiscIO.